lib/vagrant-libvirt/util/domain_flags.rb

```diff
@@ -0,0 +1,15 @@
+ # Ripped from https://libvirt.org/html/libvirt-libvirt-domain.html#types
+module VagrantPlugins
+  module ProviderLibvirt
+    module Util
+      module DomainFlags
+        # virDomainUndefineFlagsValues
+        VIR_DOMAIN_UNDEFINE_MANAGED_SAVE = 1 # Also remove any managed save
+        VIR_DOMAIN_UNDEFINE_SNAPSHOTS_METADATA = 2 # If last use of domain, then also remove any snapshot metadata
+        VIR_DOMAIN_UNDEFINE_NVRAM = 4 # Also remove any nvram file
+        VIR_DOMAIN_UNDEFINE_KEEP_NVRAM = 8 # Keep nvram file
+        VIR_DOMAIN_UNDEFINE_CHECKPOINTS_METADATA = 16 # If last use of domain, then also remove any checkpoint metadata Future undefine control flags should come here.
```

Comment:
While I completely agree with the sentiment, this approach was copied from the only other use of a `VIR_*` constant: `VIR_ERR_NO_DOMAIN`, defined in [util/error_codes.rb:50](https://github.com/vagrant-libvirt/vagrant-libvirt/blob/4feb0424cd656115d7e428e1e8c35b37a9986c38/lib/vagrant-libvirt/util/error_codes.rb#L50).

I'd guess this was done because of the somewhat unpredictable release schedule of ruby-libvirt. Maybe I'll look into patching ruby-libvirt, but judging by the spacing of their past releases it seems a bit pointless to wait on them just for a constant.

Reply:
I'd completely forgot about them, does seem excessive though as we only use one of them. I suspect they were pulled in with the assumption they would be more useful. May also have been due to not realizing the code was available as `::Libvirt::Error::ERR_NO_DOMAIN`.

It won't block this landing if ruby-libvirt don't include it in time, just be good to try since we could remove the error codes and replace with the constant directly as well.

Reply:
Upstream PR: https://gitlab.com/libvirt/libvirt-ruby/-/merge_requests/19

I wasn't able to spend any time on this issue the past few months. Unfortunately, this means that the patch missed the 0.8.0 release a month ago.